### DISCUSSION TITLE
Best way to handle timestamp with timezone

### DISCUSSION THREAD

**Discussion by amirainfo** [prisma/prisma#11915] [Q&A] [answered]
Hi there.

To mySQL . How to use Timestamp Zone ?

My code is

```prisma
model wami_users {
  id         String   @id() @default(uuid())
  name       String   @db.VarChar(255)
  email      String   @db.VarChar(255)
  password   String   @db.VarChar(255)
  created_at DateTime @default(now()) @db.Timestamp(0)
  updated_at DateTime @default(now()) @db.Timestamp(0)
  avatar     String?  @db.VarChar(255)
  avatar_url String?  @db.VarChar(255)
}
```

**Comment by nurul3101** [comment, marked as answer]
Hey @amirainfo 👋🏻 ,
From [MySQL Documentation](https://dev.mysql.com/doc/refman/8.0/en/datetime.html), it's mentioned that your timestamp will always be converted to UTC before storing:

> MySQL converts TIMESTAMP values from the current time zone to UTC for storage, and back from UTC to the current time zone for retrieval.

So for example, If your database server's timezone is CET, it will be converted to UTC first and then stored.
You can check your database server's current timezone through this [system_time_zone global variable](https://dev.mysql.com/doc/refman/8.0/en/server-system-variables.html#sysvar_system_time_zone).
If you want to convert the retrieved time from your database's timezone to a different timezone in your application code, I would suggest using libraries like day.js, which has functions that can be used to [convert timezone](https://day.js.org/docs/en/timezone/converting-to-zone)

Please let me know if you have any further questions.

**Comment by arthurlomakin11** [comment]
It's a problem with @nurul3101 solution. 
What if time zone changed? In Ukraine, for example, we have 2 time zones. First half UTC+2, second half - UTC+3.
Imagine the situation. Time 15:00 (UTC+2). In DB its 13:00 by UTC.
It's half a year later. We are doing select. 13:00 + current time zone shift (UTC + 3) = 16:00. 16:00 is not 15:00, that was first

So, it would be better to store date in string with time zone shift inside. For example, in ISO format. Like this: 2022-02-23T20:21:24+0000

MySQL stores DATETIME without timezone information. Let's say you store '2019-01-01 20:00:00' into a DATETIME field, when you retrieve that value you're expected to know what timezone it belongs to.

I think storing date in string is better, when DB don't have native support of datetime with timezone (like PostgreSQL)

**Reply by nurul3101** [reply]
If it's possible to use `DATETIME` instead of `TIMESTAMP` in your application then it would definitely make your work easier.

But if you still need to use timestamp then there would be an overhead of making sure that correct timezone is passed to the appropriate library but will still work even in case of Time zone shifts.

**Example** - In case of Eastern Time zone there are two shifts (EDT - Eastern Daylight Time [UTC - 04:00]) and (EST - Eastern standard Time [UTC - 05:00]), So while retrieving data from database and making conversion from UTC, appropriate time zone would need to be passed for correct conversion, to help you with this there are functions which you can use to check if Daylight Saving Time (DST) is in effect or not, for instance [isDST](https://momentjs.com/docs/#/query/is-daylight-saving-time/) from moment.js would help you in deciding which Timezone needs to be passed.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
